### PR TITLE
negativeLogWarning function

### PR DESCRIPTION
**Pull request by jakublala**
Added the negativeLogWarning function to detect whether any values displayed will be discarded as a result from changing the axis scale from linear to log. This will then display a warning to the user.

### CODE REVIEW

**Comment by jakublala** [comment]
Fixes #109

File: src/map/map.ts [hunk 1]
```diff
@@ -337,8 +337,22 @@ export class PropertiesMap {
 
     /** Add all the required callback to the settings */
     private _connectSettings() {
+        // Reads a warning if negative values of log are discared and not shown
+        const negativeLogWarning = (axis: AxisOptions) => {
+            if (
+                axis.scale.value === 'log' &&
+                Math.min(...(this._coordinates(axis, 0)[0] as number[])) < 0 &&
```
Comment (Luthaf) [inline src/map/map.ts:344]: This will break with large datasets. You should use `arrayMaxMin` instead

File: src/map/map.ts [hunk 2]
```diff
@@ -378,6 +393,15 @@ export class PropertiesMap {
...
                 const min = axis.min.value;
                 const max = axis.max.value;
+
+                if (name === 'xaxis') {
+                    negativeLogWarning(this._options.x);
+                } else if (name === 'yaxis') {
+                    negativeLogWarning(this._options.y);
+                } else {
+                    negativeLogWarning(this._options.z);
+                }
```
Comment (Luthaf) [inline src/map/map.ts:403]:
```suggestion
                negativeLogWarning(axis);
```

**Review by Luthaf — APPROVED** [review]
That looks good! Please squash commits together before merging! 

Here is some documentation on how/why do that, and we can do it together later today: https://www.internalpointers.com/post/squash-commits-into-one-git

**Pull request merged**